pagopa/cra-template-typescript
# Repo to CRA template

In this PR the repo changes structure to a CRA template following the [custom templates](https://create-react-app.dev/docs/custom-templates/) documentation. It is possible to test how to create a new project from the template (instructions below).

**How does this work?**
- `package.json` is just the package.json for developing the template, not that of the template created for the end user
- `template.json` is the actual package.json of the template created for the end user
- `template/*` is the content of the template, the one that will be executed with template.json as its package.json

**What to ignore (because they will be subject to a separate discussion and PR):**
- linting, formatting and TypeScript configurations
- missing libraries and their configurations
- license
- readme
- yarn vs npm as a recommended way to use the template

**Notes:**
- the gitignore file in `template/gitignore` is without the dot at the beginning. This is required. Stealing the quote from someone else on the internet, "weird, but ok"

**Assumptions made in this PR:**
- empty `.env` and `.env.local` files are not needed in the repo. In the Readme we will point to the [CRA docs page](https://create-react-app.dev/docs/adding-custom-environment-variables/) on environmental variables and give a slight edge to dotenv as a default way to add them to your repo
- `public/index.html` is configured for PagoPA (e.g. it has all the appropriate icons and manifest)
- `@pagopa/mui-italia` is installed and configured as all project will have to use it
- the `danger` and `danger-plugin-digitalcitizenship` modules and their configuration were kept as we see them in most PagoPA projects, though we are unsure whether they will be needed

**How to test it:**
- clone this repo and `cd` in the project folder 
- do a `git checkout cra-template`
- run `yarn` to install the dependencies to test the creation of a new project locally
- call `yarn create react-app my-app --template file:./` to create a new project from the template
- verify that a `my-app` project folder is created
- `cd` into `my-app`
- run `yarn start` to start the project

At this point a new Create React App with the PagoPA template should be running
